nested/blobStorageAccount.json

```diff
@@ -0,0 +1,58 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "moodleCommon": {
+            "metadata": {
+                "description": "Common Moodle values"
+            },
+            "type": "object"
+        }
+    },
+    "resources": [
+        {
+            "kind": "BlobStorage",
+            "location": "[resourceGroup().location]",
+            "name": "[parameters('moodleCommon').blobStorageAccountName]",
+            "apiVersion": "2017-06-01",
+            "sku": {
+                "name": "[parameters('moodleCommon').blobStorageAccountType]"
+            },
+            "properties": {
+                "accessTier": "Hot",
+                "encryption": {
+                    "keySource": "Microsoft.Storage",
+                    "services": {
+                        "blob": {
+                            "enabled": true
+                        },
+                        "file": {
+                            "enabled": true
+                        }
+                    }
+                },
+                "networkAcls": {
+                    "bypass": "AzureServices",
+                    "defaultAction": "Allow",
+                    "ipRules": [],
+                    "virtualNetworkRules": []
+                },
+                "supportsHttpsTrafficOnly": true
+            },
+            "type": "Microsoft.Storage/storageAccounts"
+        }
+    ],
+    "variables": {
+        "documentation1": "This sub-template creates a blob storage account.  It expects certain values in the 'common' datastructure.",
+        "documentation2": " blobStorageAccountName - name of storage account",
+        "documentation3": " blobStorageAccountType - type of storage account",
+        "scriptUri": "[concat(parameters('moodleCommon').ScriptLocation,parameters('moodleCommon').moodleInstallScriptFilename)]",
+        "blobStorageAccountId": "[concat(resourceGroup().id,'/providers/Microsoft.Storage/storageAccounts/', parameters('moodleCommon').blobStorageAccountName)]"
+    },
+    "outputs": {
+        "blobStorageAccountKey": {
+            "value": "[listKeys(variables('blobStorageAccountId'), '2017-06-01').keys[0].value]",
+            "type": "string"
```

Comment:
secureString

Reply:
Done

Reply:
Ditto (removed the string secrets outputs/params all together).